src/SFA.DAS.Reservations.Functions.Reservations.AcceptanceTests/Steps/ReservationCreatedSteps.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using SFA.DAS.Reservations.Data;
+using SFA.DAS.Reservations.Domain.Accounts;
+using SFA.DAS.Reservations.Domain.Entities;
+using SFA.DAS.Reservations.Domain.Notifications;
+using SFA.DAS.Reservations.Domain.ProviderPermissions;
+using SFA.DAS.Reservations.Domain.Reservations;
+using SFA.DAS.Reservations.Messages;
+using TechTalk.SpecFlow;
+using Reservation = SFA.DAS.Reservations.Domain.Reservations.Reservation;
+
+namespace SFA.DAS.Reservations.Functions.Reservations.AcceptanceTests.Steps
+{
+    [Binding]
+    public class ReservationCreatedSteps : StepsBase
+    {
+        public ReservationCreatedSteps(TestServiceProvider serviceProvider, TestData testData) : base(serviceProvider, testData)
+        {
+        }
+        
+        [Given(@"I have a reservation ready for creation")]
+        public void GivenIHaveAReservationReadyForCreation()
+        {
+            TestData.Reservation.Status = (short)ReservationStatus.Pending;
+        }
+        
+        [When(@"a create reservation event is triggered")]
+        public void WhenACreateReservationEventIsTriggered()
+        {
+            var permissionRepository = Services.GetService<IProviderPermissionRepository>();
+            var mockPermissionRepository = Mock.Get(permissionRepository);
+
+            var providerPermission = new ProviderPermission
+            { AccountId = 1, AccountLegalEntityId = 1, CanCreateCohort = true, ProviderId = 1 };
+
+            mockPermissionRepository.Setup(x => x.GetAllForAccountLegalEntity(TestData.AccountLegalEntity.AccountLegalEntityId))
+                .Returns(new List<ProviderPermission> { providerPermission });
```

Comment:
Not sure if its worth moving this into the TestData setup?

Reply:
Done